configs/components/oifs/oifs.yaml

```diff
@@ -856,6 +887,56 @@ choose_version:
                 add_outdata_files:
                         ICMUA: ICMUA
 
+
+# If 1, use Coddington spectrum which reduced stratospheric warm bias
+# Only available in 43r3-v2
+choose_solarspectrum:
+        True:
+                add_namelist_changes:
+                        fort.4:
+                                NAERAD:                         
+                                        NSOLARSPECTRUM: 1
+
+# There are two different O3 concentrations in OpenIFS:
+# One is used for output, and one that is used by radiation scheme
+# By default, the two are NOT the same. 
+# By default, prognostic O3 from the Cariolle scheme is written to output
+# but climatology is used by radiation scheme
+# 
+# Here the scheme can be changed in two ways:
+# cmip6: Use O3 from CMIP6 (input4MIP) for picontrol, historical or scenarios
+# cariolle: Use the prognostic O3 from Cariolle in radiation
+choose_o3_scheme:
+        cmip6:                
+                # For CMIP6 O3 we set
+                # LCMIP6O3 to true
+                # LEO3VAR to true
+                # LEPO3RA to false (otherwise prognostic O3 is used)
+                add_namelist_changes:
+                        fort.4:
+                                NAMGFL:
+                                        YO3_NL%LGP: '.true.' # O3 is a tracer
+                                NAEPHY:
+                                        LEO3CH: '.true.' # turn on O3 scheme
+                                NAERAD:
+                                        LEPO3RA: '.false.' # dont use progn O3 in radiation
+                                        LCMIP6O3: '.true.' # use CMIP6 O3
+                                        LEO3VAR: '.true.'  # allow time-varying O3 (not climatology)
+                                        NGHGRAD: 21        # dont overwrite by progn O3
+
+        cariolle:
+                # For Cariolle
+                add_namelist_changes:
+                        fort.4:
+                                NAMGFL:
+                                        YO3_NL%LGP: '.true.'
+                                NAEPHY:
+                                        LEO3CH: '.true.'
+                                NAERAD:
+                                        LEPO3RA: '.true.' # use progn O3 in radiation
+                                        LCMIP6O3: '.true.'                                        
+                                        NGHGRAD: 20       # overwrite O3 by progn O3
+         
```

Comment:
Hm, maybe we should have a chat about this. I've been unsure how to deal with "solarspectrum" myself since it only works for OpenIFS 43r3v2 (the namelist variable does not exist in v1). 

My thoughts are: 
1) I don't like the idea of namelist templates, and I thought the whole point with ESM-Tools would be to use one single template, and then control everything via switches. 
We use templates for the xml files in XIOS (since switches don't work) and it's really terrible! 

2) I've tried to make it so that setting any of these switches to "default" or "0" or "no" would give you the default OpenIFS behaviour. For example: Unless you set use_ocean_currents, it will be "0", which makes "LECURR = false" and ocean surface velocity never goes into the computation of surface wind stress, as is ECMWF default. Similarly "o3_scheme: default" gives you ECMWF default behaviour (which is not CMIP compatible). 

Let's have a chat about this. Friday 7th at 9am? 

Best wishes
Joakim

Reply:
Hi Joakim, that time would work for me.

Reply:
Same here. I'm available Friday at 9am. Let's hope that @JanStreffing has time too.